nginx/datadog_checks/nginx/nginx.py

```diff
@@ -90,14 +91,74 @@ def check(self, _):
             except Exception as e:
                 self.log.error('Could not submit metric: %s: %s', repr(row), e)
 
+    def _get_enabled_endpoints(self):
+        """
+        Dynamically determines which NGINX endpoints are enabled and Datadog supports getting metrics from
+        by querying the NGINX APIs that list availabled endpoints. If an error is encountered,
+        then it falls back to query all of the known endpoints available in the given NGINX Plus version.
+        """
+        available_endpoints = set()
+        base_url = "/".join([self.url, self.plus_api_version])
+        http_url = "/".join([self.url, self.plus_api_version, "http"])
+        stream_url = "/".join([self.url, self.plus_api_version, "stream"])
```

Comment:
we should use urljoin for constructing these https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urljoin

Reply:
Updated, let me know if it looks better now